src/DssSpellCollateral.sol

```diff
@@ -28,60 +61,260 @@ contract DssSpellCollateralAction {
     // $ bc -l <<< 'scale=27; e( l(1.08)/(60 * 60 * 24 * 365) )'
     //
     // A table of rates can be found at
-    //    https://ipfs.io/ipfs/QmTRiQ3GqjCiRhh1ojzKzgScmSsiwQPLyjhgYSxZASQekj
+    //    https://ipfs.io/ipfs/QmX2QMoM1SZq2XMoTbMak8pZP86Y2icpgPAKDjQg4r4YHn
     //
 
+    uint256 constant ZERO_PCT_RATE           = 1000000000000000000000000000;
+    uint256 constant ZERO_ZERO_FIVE_PCT_RATE = 1000000000015850933588756013;
+
+    // --- Math ---
+    uint256 public constant WAD = 10**18;
+    uint256 public constant RAY = 10**27;
+    uint256 public constant RAD = 10**45;
+
+    // -- RWA008 MIP21 components --
+    address constant RWA008                    = 0x7907FB7f6aa0ebe363BC6966c4132849b7B560F8;
+    address constant MCD_JOIN_RWA008_A         = 0x8EeE937787DFD6A82ED1F1d11d9db7e2dC3a5884;
+    address constant RWA008_A_URN              = 0x7b7061164858B69cFc3B0a3B8b630C9279c20de5;
+    address constant RWA008_A_URN_CLOSE_HELPER = 0xDCd1D2879A309785b4E40b6Fb3FbE36D315bC597;
```

Comment:
Where is the code for this contract located?

Reply:
https://github.com/clio-finance/socgen-ofh-onboarding/blob/master/src/RwaUrnCloseHelper.sol